tests/test_network.py

```diff
@@ -0,0 +1,96 @@
+"""Test module for `network.py`"""
+
+import unittest
+import unittest.mock
+
+import network
+
+
+class TestNetwork(unittest.TestCase):
+    """Test class for `Network`"""
+
+    def test_connected(self):
+        """Test that `connected` correctly reports connection status"""
+        networktables_mock = unittest.mock.Mock()
+        networktables_mock.isConnected.side_effect = [True, False]
+
+        network_instance = network.Network(networktables_mock, None, None)
```

Comment:
Nice work! You really went to town on these tests, and an effective use of mocks to boot!